premiere/src/utils/video_sequence_parser.rs

```diff
@@ -0,0 +1,79 @@
+use crate::*;
+use pr_sys::*;
+use std::collections::HashMap;
+
+#[derive(Debug, Clone)]
+pub struct ClipOperator {
+    pub node_id: i32,
+    pub node_type: String,
+    pub hash: prPluginID,
+    pub flags: i32,
+}
+
+pub type ClipOperatorsMap = HashMap<i32, ClipOperator>;
+
+pub struct VideoSequenceParser {
+    segment_suite: suites::VideoSegment,
+}
+
+impl VideoSequenceParser {
+    pub fn new() -> Result<Self, Error> {
+        Ok(Self {
+            segment_suite: suites::VideoSegment::new()?,
+        })
+    }
+
+    pub fn parse_clip_operators(&self, clip_node_id: i32) -> Result<ClipOperatorsMap, Error> {
+        let clip_node_operators = self.segment_suite.node_operator_count(clip_node_id)?;
+        let mut operators_map: ClipOperatorsMap = HashMap::new();
+
+        for operator_node_index in 0..clip_node_operators {
+            let operator_node_id = self
+                .segment_suite
+                .acquire_operator_node_id(clip_node_id, operator_node_index)?;
+
+            let (operator_node_type, operator_node_hash, operator_node_flags) =
+                self.segment_suite.node_info(operator_node_id)?;
+
+            log::debug!(
+                "Clip Operator: {:?} {:?} {:?}",
+                operator_node_type,
+                operator_node_hash,
+                operator_node_flags
+            );
```

Comment:
Instead of all these `log::debug!` calls, I think the `parse_clip_operators` should return a structute which has all these fields you print here, but the struct impls the `Debug` trait,. The end-user should just `dbg!(parse_clip_operators(...)` if they desire, or use the values in other way

Reply:
Or return a tuple of the current `Ok` and namely `struct`, i.e. `Result<(ClipOperatorsMap, ParseClipOperatorStatus), Error>`